### DISCUSSION TITLE
GLS fast local search or greedy local search?

### DISCUSSION THREAD

**Discussion by aloumpas** [google/or-tools#2976] [Routing (and legacy CP) questions]
Hi!

I would like to ask which local search strategy the implementation of guided local search is using. Fast local search or greedy local search?

If greedy local search is used is there any way to enable fast local search?

I'm using or-tools v7.1

Thank you.

**Comment by Mizux** [comment]
Did you see https://developers.google.com/optimization/routing/routing_options#local_search_options ?

**Comment by aloumpas** [comment]
Hi, Yes i had read the routing options and i am not referring to them.

To better understand in what i'm referring to please have a look to the following paper: https://www.researchgate.net/publication/225939903_Guided_Local_Search

In page 7 at third paragraph is being analyzed the rough idea about fast local search. Fast local search splits the problem into smaller sub-neighborhoods and keep an active/inactive flag for each of them. During the local search the algorithm only looks to the active sub-neighborhoods. In contrast greedy local search makes a full search so,  much more sub-neighborhoods are being evaluated.

While searching for information about the implementation being used for guided local search i found the following method:

https://github.com/google/or-tools/blob/dbac8c324d1fa3d37ed992174aa4c0fcc48cd673/ortools/constraint_solver/constraint_solver.h#L2815-L2822

Enabling or disabling this seems to have no effect on the results. So i would like to know which algorithm has been implemented on GLS.
